streaming/replication/Utils.h

```diff
@@ -0,0 +1,161 @@
+#ifndef REPLICATION_UTILS_H
+#define REPLICATION_UTILS_H
+
+#include "replication-private.h"
+
+namespace replication {
+
+/*
+ * Wraps a standard std::mutex and enables/disables cancelability
+ * in lock/unlock operations.
+ */
+class Mutex {
```

Comment:
Is this object needed?
The standard mutex functions of netdata, already do this.

Reply:
Resolving this because it belongs in the greater C vs. C++ comment you made and we need to have that discussion.

Reply:
FWIW, it's needed to enable scope-based locks that automatically lock/unlock mutexes based on scope.